kaggle/api/kaggle_api_extended.py

```diff
@@ -1316,24 +1349,53 @@ def dataset_metadata_update(self, dataset, path):
     meta_file = self.get_dataset_metadata_file(effective_path)
     with open(meta_file, 'r') as f:
       metadata = json.load(f)
-      updateSettingsRequest = DatasetUpdateSettingsRequest(
-          title=metadata['title'],
-          subtitle=metadata['subtitle'],
-          description=metadata['description'],
-          is_private=metadata['isPrivate'],
-          licenses=[License(name=l['name']) for l in metadata['licenses']],
-          keywords=metadata['keywords'],
-          collaborators=[
-              Collaborator(username=c['username'], role=c['role'])
-              for c in metadata['collaborators']
-          ],
-          data=metadata['data'])
-      result = self.process_response(
-          self.metadata_post_with_http_info(owner_slug, dataset_slug,
-                                            updateSettingsRequest))
-      if (len(result['errors']) > 0):
-        [print(e['message']) for e in result['errors']]
-        exit(1)
+      with self.build_kaggle_client() as kaggle:
+        settings = DatasetSettings()
+        settings.title = metadata['title']
+        settings.subtitle = metadata['subtitle']
+        settings.description = metadata['description']
+        settings.is_private = metadata['isPrivate']
+        settings.licenses = [
+            self.new_license(l['name']) for l in metadata['licenses']
+        ]
+        settings.keywords = metadata['keywords']
+        settings.collaborators = [
+            self.new_collaborator(c['username'], c['role'])
+            for c in metadata['collaborators']
+        ]
+        settings.data = [self.new_metadata() for m in metadata['data']]
+        request = ApiUpdateDatasetMetadataRequest()
+        request.settings = settings
+        request.owner_slug = owner_slug
+        request.dataset_slug = dataset_slug
+        response = kaggle.datasets.dataset_api_client.update_dataset_metadata(
+            request)
+
+      if len(response.errors) > 0:
+        [print(e) for e in response.errors]
+      return response
+
+  def new_license(self, name):
+    slicense = SettingsLicense()
+    slicense.name = name
+    return slicense
+
+  def new_collaborator(self, name, role):
+    collab = UserRole()
+    collab.username = name
+    collab.role = role
+    return collab
+
+  def new_metadata(self, data):
+    if len(data) == 0:
+      return None
+    md = DatasetSettingsFile()
+    # TODO Verify this is correct.

```

Comment:
This is a little concerning... did you verify it?

Reply:
Not yet. That's why there's a TODO; I don't want to forget it.

Reply:
Ok, but this code is being merged to main, which is where we cut releases from.  I'm worried about turning `main` into something that can't be released.  Is there a contingency plan?  Do we need another branching strategy, like maybe a long-running branch for the transition to kapigen?  AFAIK, everything that's gone in so far could be released, despite the fact that we'd prefer not to have a mixed kapigen/swagger strategy in prod.

Reply:
Deleted.